primap2/pm2io/_data_reading.py

```diff
@@ -191,37 +219,45 @@ def convert_long_dataframe_if(
     if add_coords_cols:
         check_overlapping_specifications_add_cols(coords_cols, add_coords_cols)
 
-    filter_data(data_long, filter_keep, filter_remove)
+    # make a copy to keep input dataframe unchanged
+    data_copy = data_long.copy(deep=True)
+
+    filter_data(data_copy, filter_keep, filter_remove)
 
     add_dimensions_from_defaults(
-        data_long, coords_defaults, additional_allowed_coords=["time"]
+        data_copy, coords_defaults, additional_allowed_coords=["time"]
     )
 
     naming_attrs = rename_columns(
-        data_long, coords_cols, add_coords_cols, coords_defaults, coords_terminologies
+        data_copy, coords_cols, add_coords_cols, coords_defaults, coords_terminologies
     )
-
     attrs.update(naming_attrs)
 
+    # replace the NA_Values
+    if convert_nan:
+        repl_dict = dict(zip(NA_VALUES, list(np.full(len(NA_VALUES), np.nan))))
+        data_copy["data"] = data_copy["data"].replace(repl_dict)
+        data_copy["data"] = pd.to_numeric(data_copy["data"], errors="coerce")
```

Comment:
duplicated in `long` and `wide`, I moved it to a separate function.

Reply:
OK, great